### PR TITLE
[FRAF-752] - TS fixes related to guild week

### PR DESCRIPTION
**Pull request by eniskraasniqi1**
### Link to issue
https://teamleader.atlassian.net/browse/FRAF-752

### Description

We decided to drop all used types and instead use string literals or unions. Now we use only constant types that come from constants folder (more for the generalised types like colors, tint, size), and all other cases inline typing with string literals.
Also this PR fixes issues related to Guild week!

To manual check this:

- Link ui library locally with core branch (**ui-library-type-updates**)
- Confirm that there are no errors left when running ts-check

### CODE REVIEW

File: src/components/datepicker/DatePickerInput.tsx [hunk 1]
```diff
@@ -47,10 +47,10 @@ interface DatePickerInputProps extends Omit<BoxProps, 'size' | 'onChange'> {
 }
 
 interface DayPickerProps {
-  numberOfMonths: number;
-  showOutsideDays: boolean;
-  showWeekNumbers: boolean;
-  withMonthPicker: boolean;
+  numberOfMonths?: number;
```
[the remark this answers is not in the document]
Comment (eniskraasniqi1) [reply src/components/datepicker/DatePickerInput.tsx:50]: I think they can be optional, also if you check Datepicker.tsx two of these props are optional there too. But probably @qubis741 has a better answer.

File: src/components/popover/popover.stories.tsx [hunk 2]
```diff
@@ -32,6 +31,11 @@ const backdrops = ['transparent', 'dark'];
 const directions = ['north', 'south', 'east', 'west'];
 const positions = ['start', 'center', 'end'];
 
+type Color = 'aqua' | 'gold' | 'mint' | 'neutral' | 'ruby' | 'teal' | 'violet';
```
Comment (stefaandevylder) [inline src/components/popover/popover.stories.tsx:34]: If you are doing the `COLORS` everywhere, why not here?

File: src/components/marketingButton/MarketingButton.tsx [hunk 3]
```diff
@@ -19,23 +15,23 @@ interface MarketingButtonProps extends Omit<BoxProps, 'ref' | 'element'> {
...
+  iconPlacement?: 'left' | 'right';
   /** Callback function that is fired when mouse leaves the component. */
   onMouseLeave?: (event: React.MouseEvent) => void;
   /** Callback function that is fired when the mouse button is released. */
   onMouseUp?: (event: React.MouseEvent) => void;
   /** If true, component will show a loading spinner instead of label or children. */
   processing?: boolean;
   /** Size of the button. */
-  size?: Size;
+  size?: 'tiny' | 'small' | 'medium' | 'large';
```
Comment (farazatarodi) [inline src/components/marketingButton/MarketingButton.tsx:34]: Can't use the constant here?

File: src/components/marketingLockBadge/MarketingLockBadge.tsx [hunk 4]
```diff
@@ -7,11 +7,9 @@ import { IconLockSmallFilled } from '@teamleader/ui-icons';
 import { GenericComponent } from '../../@types/types';
 import { BoxProps } from '../box/Box';
 
-type Size = 'small' | 'medium';
-
 interface MarketingLockBadgeProps extends Omit<BoxProps, 'ref'> {
   className?: string;
-  size?: Size;
+  size?: 'small' | 'medium';
```
Comment (farazatarodi) [inline src/components/marketingLockBadge/MarketingLockBadge.tsx:12]: And here.

File: src/components/marketingStatusLabel/MarketingStatusLabel.tsx [hunk 5]
```diff
@@ -8,12 +8,10 @@ import { UITextBody, UITextSmall } from '../typography';
 import { GenericComponent } from '../../@types/types';
 import { BoxProps } from '../box/Box';
 
-type Size = 'small' | 'medium';
-
 interface MarketingStatusLabelProps extends Omit<BoxProps, 'ref'> {
   children?: ReactNode;
   fullWidth?: boolean;
-  size?: Size;
+  size?: 'small' | 'medium';
```
Comment (farazatarodi) [inline src/components/marketingStatusLabel/MarketingStatusLabel.tsx:14]: And here.

File: src/components/marketingTab/MarketingTab.tsx [hunk 6]
```diff
@@ -7,14 +7,12 @@ import { Heading4, Heading5 } from '../typography';
...
 
-type Size = 'small' | 'medium';
-
 interface MarketingTabProps extends Omit<BoxProps, 'ref'> {
   active?: boolean;
   children: ReactNode;
   className?: string;
   onClick?: (event: React.MouseEvent) => void;
-  size?: Size;
+  size?: 'small' | 'medium';
```
Comment (farazatarodi) [inline src/components/marketingTab/MarketingTab.tsx:15]: You get the point 🤣

File: src/components/popover/Popover.tsx [hunk 7]
```diff
@@ -32,9 +27,9 @@ interface PopoverProps extends Omit<BoxProps, 'ref'> {
   /** The class names for the wrapper to apply custom styling. */
   className?: string;
   /** The background colour of the Popover. */
-  color?: Color;
+  color?: 'aqua' | 'gold' | 'mint' | 'neutral' | 'ruby' | 'teal' | 'violet';
```
Comment (farazatarodi) [inline src/components/popover/Popover.tsx]: Why are we not using `COLORS` here?

File: src/components/iconButton/iconButton.stories.spec.tsx [hunk 8]
```diff
@@ -14,9 +15,19 @@ const Wrapper = ({ children, inverse = false }: { children: ReactNode; inverse:
     {children}
   </div>
 );
+type Sizes = 'small' | 'medium' | 'large';
 const sizes: Sizes[] = ['small', 'medium', 'large'];
-const colors: Colors[] = ['neutral', 'white', 'mint', 'violet', 'ruby', 'gold', 'aqua', 'teal'];
-const tints: Tints[] = ['lightest', 'light', 'normal', 'dark', 'darkest'];
+const colors: (typeof COLORS[number] | 'white')[] = [
+  'neutral',
+  'white',
```
Comment (farazatarodi) [inline src/components/iconButton/iconButton.stories.spec.tsx:22]: There is no `white` color. We have a `neutral` color and 'lightest` tint.

**Review by farazatarodi — CHANGES_REQUESTED** [review]
See my comments and also use the functions we have instead of `Exclude` for types.

File: src/components/popover/Popover.tsx (hunk 7 above)
Comment (eniskraasniqi1) [reply src/components/popover/Popover.tsx]: This is outdated, fixed this 👍

File: src/components/iconButton/iconButton.stories.spec.tsx (hunk 8 above)
Comment (eniskraasniqi1) [reply src/components/iconButton/iconButton.stories.spec.tsx:22]: So remove white from all the places. Saw multiple places where we used it.

**Review by qubis741 — DISMISSED** [review]

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

**Review by stefaandevylder — DISMISSED** [review]

**Review by lowiebenoot — APPROVED** [review]

**Pull request merged**